src/main/java/org/apache/commons/csv/ExtendedBufferedReader.java

```diff
@@ -92,7 +93,8 @@ public int read(final char[] buf, final int offset, final int length) throws IOE
                     if (CR != (i > 0 ? buf[i - 1] : lastChar)) {
                         eolCounter++;
                     }
-                } else if (ch == CR) {
+                } else if ((ch == CR)
+                        || (ch == END_OF_STREAM && (i > 0 ? buf[i - 1] : lastChar) != LF && (i > 0 ? buf[i - 1] : lastChar) != CR)) {
```

Comment:
Can we make the code more self-explaining? 

* `i > 0 ? buf[i - 1`is used three times in this code snippet